deploy/examples/nfs.yaml

```diff
@@ -76,3 +95,34 @@ spec:
 #   replicated:
 #     size: 3
 #     requireSafeReplicaSize: true
+# ---
+# # Example configmap for providing sssd.conf file to the SSSD sidecar
+# apiVersion: v1
+# kind: ConfigMap
+# metadata:
+#   name: my-nfs-sssd-config
+#   namespace: rook-ceph # namespace:cluster
+# data:
+#   sssd.conf: |
+#     [sssd]
+#     # Only the nss service is required for the SSSD sidecar.
+#     services = nss
+#     domains = default
+#     config_file_version = 2
+#
+#     [nss]
+#     filter_users = root
+#
+#     [domain/default]
+#     id_provider = ldap
+#     ldap_uri = ldap://server.address
+#     ldap_search_base = dc=company,dc=net
+#     ldap_default_bind_dn = cn=admin,dc=company,dc=net
+#     ldap_default_authtok_type = password
```

Comment:
The password will be in plain text in the configmap?

Reply:
Erm... yes. :(
Good point.

Reply:
So this can be set to `obfuscated_password` and then an obfuscated password can be provided. One of the follow-up tasks for the NFS work is to support CA certs for authentication in the future. IMO, supporting ConfigMap as a source for sssd.conf still makes sense. And I deliberately designed the API such that other auth methods can be added later, like if we want to add Secret as a follow-up.

Reply:
Right. And with the vault agent injector, I think it's pretty easy to add something like a CA cert or config file as well, both of which would be user-created. No changes would be needed for Rook there at all.

Reply:
Maybe I should remove the requirement that sssdConfigFile have a spec so we can support vault or other secret injectors?